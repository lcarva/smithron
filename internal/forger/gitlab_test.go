package forger

import (
	"context"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestGitLab(t *testing.T) {
	p := Plan{
		Name:    "hello-world",
		Image:   "docker.io/busybox:latest",
		Command: []string{"echo", "Hello, World!"},
		Env: map[string]string{
			"GREETING": "Hello",
		},
	}

	ctx := context.Background()

	tf, err := GetForgerForTarget("gitlab")
	require.NoError(t, err)
	require.IsType(t, GitLabForger{}, tf)

	got, err := tf.Forge(ctx, p)
	require.NoError(t, err)
	snaps.MatchYAML(t, got)
}
