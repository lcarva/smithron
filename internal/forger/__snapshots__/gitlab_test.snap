
[TestGitLab - 1]
hello-world:
  image:
    entrypoint:
    - echo
    - Hello, World!
    name: docker.io/busybox:latest
  variables:
    GREETING: Hello

---
